metabase/metabase
# Add warning message when editing an existing SQL question

_**Blocked by https://github.com/metabase/metabase/pull/30290, I'm creating the QueryBuilder test environment there and this will use the same one**_

### Description

Adds "Are you sure?" prompt when a user tries to refresh/close/navigate away from the current page, after the user has edited an existing SQL question.


### How to verify

1. Go to an existing SQL question
2. Edit the query (ex: click `Open Editor` and add/delete a character in the SQL query)
3. Try to refresh the page, you should see a message similar to that of the screenshot below.

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30294)
